ELWebServiceTests/WebServiceTests.swift

```diff
@@ -13,136 +13,226 @@ import XCTest
-    }
-    
     func test_get_createsGETRequest() {
         let service = WebService(baseURLString:  "http://httpbin.org/")
         let session = RequestRecordingSession()
         service.session = session
-        
+
         let task = service.GET("/get")
         task.resume()
-        
+
         let recordedRequest = session.recordedRequests.first?.urlRequestValue
         XCTAssertNotNil(recordedRequest)
-        
+
         let method = recordedRequest?.httpMethod
         XCTAssertNotNil(method)
-        
+
         XCTAssertEqual(method!, "GET")
     }
-    
+
     func test_post_createPOSTRequest() {
         let service = WebService(baseURLString:  "http://httpbin.org/")
         let session = RequestRecordingSession()
         service.session = session
-        
+
         let task = service.POST("/post")
         task.resume()
-        
+
         let recordedRequest = session.recordedRequests.first?.urlRequestValue
         XCTAssertNotNil(recordedRequest)
-        
+
         let method = recordedRequest?.httpMethod
         XCTAssertNotNil(method)
-        
+
         XCTAssertEqual(method!, "POST")
     }
-    
+
     func test_delete_createDELETERequest() {
         let service = WebService(baseURLString:  "http://httpbin.org/")
         let session = RequestRecordingSession()
         service.session = session
-        
+
         let task = service.DELETE("/delete")
         task.resume()
-        
+
         let recordedRequest = session.recordedRequests.first?.urlRequestValue
         XCTAssertNotNil(recordedRequest)
-        
+
         let method = recordedRequest?.httpMethod
         XCTAssertNotNil(method)
-        
+
         XCTAssertEqual(method!, "DELETE")
     }
-    
+
     func test_head_createHEADRequest() {
         let service = WebService(baseURLString:  "http://httpbin.org/")
         let session = RequestRecordingSession()
         service.session = session
-        
+
         let task = service.HEAD("/head")
         task.resume()
-        
+
         let recordedRequest = session.recordedRequests.first?.urlRequestValue
         XCTAssertNotNil(recordedRequest)
-        
+
         let method = recordedRequest?.httpMethod
         XCTAssertNotNil(method)
-        
+
         XCTAssertEqual(method!, "HEAD")
     }
-    
+
     func test_put_createPUTRequest() {
         let service = WebService(baseURLString:  "http://httpbin.org/")
         let session = RequestRecordingSession()
         service.session = session
-        
+
         let task = service.PUT("/put")
         task.resume()
-        
+
         let recordedRequest = session.recordedRequests.first?.urlRequestValue
         XCTAssertNotNil(recordedRequest)
-        
+
         let method = recordedRequest?.httpMethod
         XCTAssertNotNil(method)
-        
+
         XCTAssertEqual(method!, "PUT")
     }
 }
 
 // MARK: - absoluteURLString
 
 extension WebServiceTests {
-    func test_absoluteURLString_constructsValidAbsoluteURL() {
```

Comment:
test_absoluteURLString_constructsValidAbsoluteURL is equivalent to test_absoluteURLString_baseTrailingSlash_relativeLeadingSlash.

test_absoluteURLString_constructsValidURLWhenPathIsAbsoluteURL is very similar to test_absoluteURLString_relativeHasScheme. But I will add that one back as it is slightly different.